src/history.js

```diff
@@ -260,16 +260,17 @@ function applyTransaction(history, state, tr, options) {
 
   if (tr.getMeta(closeHistoryKey)) history = new HistoryState(history.done, history.undone, null, 0)
 
-  let appended = tr.getMeta("appendedTransaction")
+  let rootTr = tr.getMeta("appendedTransaction")
   if (tr.steps.length == 0) {
     return history
-  } else if ((appended || tr).getMeta("addToHistory") !== false) {
+  } else if (tr.getMeta("addToHistory") !== false) {
```

Comment:
I'd like to keep the behavior where transactions appended to a non-history root transaction aren't added to history either. What is the reason for this change?

Reply:
As far as I can see this is the same thing. A transaction's `appendedTransaction` [can only be itself](https://github.com/ProseMirror/prosemirror-state/blob/545244357dbc12cc1f340cd3f8ad75dc7472a803/src/state.js#L131) (if at all). So `appended` is either falsey [or `=== tr`](https://github.com/RichieAHB/prosemirror-history/blob/22f773640ce79c91f328d14ef20de8361b2fb7f0/src/history.js#L263), unless there's somewhere else that `appendedTransaction` is being set that I've missed?

Reply:
> A transaction's appendedTransaction can only be itself (if at all).

Right, but that's a bug (the behavior as you fixed it in the other PR was what I was _trying_ to implement there)